test/values.test.js

```diff
@@ -501,12 +538,176 @@ describe('Values', () => {
+      }),
+      'Query(Lambda("_", Match(Index("idx"), 10)))'
+    )
+
+    // Filter an array
+    assertPrint(
+      new Query({
+        collection: [1, 2, 3],
+        api_version: '3',
+        filter: {
+          expr: {
+            equals: [0, { modulo: [{ var: 'i' }, 2] }],
+          },
+          lambda: 'i',
+        },
+      }),
+      'Query(Filter([1, 2, 3], Lambda("i", Equals(0, Modulo(Var("i"), 2)))))'
+    )
+
+    // Filter a page
+    assertPrint(
+      new Query({
+        filter: {
+          lambda: 'i',
+          expr: {
+            containsstr: { select: 'name', from: { get: { var: 'i' } } },
+            search: '-',
+          },
+        },
+        collection: { paginate: { databases: null } },
+      }),
+      'Query(Filter(Paginate(Databases()), Lambda("i", ContainsStr(Select("name", Get(Var("i"))), "-"))))'
+    )
+
+    // Select
+    assertPrint(
+      new Query({
+        from: {
+          object: {
+            favorites: {
+              object: {
+                foods: ['crunchings', 'munchings', 'lunchings'],
+              },
+            },
+          },
+        },
+        select: ['favorites', 'foods', 1],
+      }),
+      'Query(Select(["favorites", "foods", 1], {favorites: {foods: ["crunchings", "munchings", "lunchings"]}}))'
+    )
+
+    // Map an Array
+    assertPrint(
+      new Query({
+        map: { lambda: 'x', expr: { add: [{ var: 'x' }, 1] } },
+        collection: [1, 2, 3],
+      }),
+      'Query(Map([1, 2, 3], Lambda("x", Add(Var("x"), 1))))'
+    )
+
+    // Map a Page
+    new assertPrint(
+      new Query({
+        map: {
+          expr: { get: { var: 'x' } },
+          lambda: 'x',
+        },
+        collection: { paginate: { databases: null } },
+      }),
+      'Query(Map(Paginate(Databases()), Lambda("x", Get(Var("x")))))'
+    )
+
+    // Foreach
+    assertPrint(
+      new Query({
+        collection: { paginate: { documents: { collection: 'orders' } } },
+        foreach: {
+          lambda: 'doc',
+          expr: {
+            update: { var: 'doc' },
+            params: { object: { data: { object: { createdBy: 'Ryan' } } } },
+          },
+        },
+      }),
+      `Query(Foreach(Paginate(Documents(Collection("orders"))), Lambda("doc", Update(Var("doc"), {data: {createdBy: "Ryan"}}))))`
+    )
+
+    // If
+    assertPrint(
+      new Query({
+        else: 'was false',
+        then: 'was true',
+        if: true,
+      }),
+      `Query(If(true, "was true", "was false"))`
+    )
+
+    // Call
+    assertPrint(
+      new Query({
+        arguments: 2,
+        call: {
+          function: 'double',
+        },
+      }),
+      `Query(Call(Function("double"), 2))`
+    )
+
+    assertPrint(
+      new Query({
+        arguments: [1, 2, 3],
+        call: {
+          function: 'double',
+        },
+      }),
+      `Query(Call(Function("double"), [1, 2, 3]))`
+    )
+
+    assertPrint(
+      new Query({
+        arguments: {
```

Comment:
that argument is not correct, objects are wrapped in the protocol, the correct would be
```
new Query({
  arguments: {
    object: {
      one: 1,
      two: 2,
      three: 3
    }
  },
  call: {
    function: 'double'
  }
})
```

Reply:
Ah, okay. Thank you. I was missing the `object` 🤦